portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,24 +42,33 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    ArrayList<String> commentSection = new ArrayList<String>();
+    List<Map.Entry<String,String>> commentSection = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
-        String text = (String) entity.getProperty("text");
-        commentSection.add(text);
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("text");
+      Map.Entry<String,String> nameAndComment=new AbstractMap.SimpleEntry<>(name,text);
```

Comment:
You shouldn't need to import/reference `AbstractMap` unless you somehow end up having to deal with the internals of `Map`. It looks like you're trying to get a `List` of comments here, which you're representing as a pair of name + text. I recommend you create a small `Comment` class to use here instead. GSON should be able to serialize it all the same, except now you'll have named fields in your JSON output rather than the more generic "key" and "value" names.